thumbor/handlers/__init__.py

```diff
@@ -281,7 +281,7 @@ def finish_request(self, context, result_from_storage=None):
             context.config.RESULT_STORAGE_STORES_UNSAFE or not context.request.unsafe)
 
         def inner(future):
-            results, content_type = future.result()
+            results, content_type = (future.result()[0], BaseEngine.get_mimetype(future.result()[0]))
```

Comment:
@okor by moving the BaseEngine.get_mimetype to inside the _load_results, you would avoid getting the content_type twice from different sources: https://github.com/okor/thumbor/blob/master/thumbor/handlers/__init__.py#L246
And I think that the callback runs synchronously.